hedera-node/src/main/java/com/hedera/services/fees/charging/NarratedCharging.java

```diff
@@ -20,21 +20,27 @@
  * ‍
  */
 
-import com.hederahashgraph.api.proto.java.AccountID;
+import com.hedera.services.utils.TxnAccessor;
+import com.hederahashgraph.fee.FeeObject;
 
 /**
- * Defines a type able to determine if a given payer can afford
- * to pay a given amount.
- *
- * @author Michael Tinker
+ * Defines the checks and charging actions we need to apply the Services fee policy.
  */
-@FunctionalInterface
-public interface BalanceCheck {
-	/**
-	 * Flags if the given payer can afford the given amount.
-	 * @param payer a payer
-	 * @param amount an amount
-	 * @return if the payer can afford the amount
-	 */
-	boolean canAfford(AccountID payer, long amount);
+public interface NarratedCharging {
+	void setFees(FeeObject fees);
+	void resetForTxn(TxnAccessor accessor, long submittingNodeId);
+
+	boolean canPayerAffordAllFees();
+	boolean canPayerAffordNetworkFee();
+	boolean canPayerAffordServiceFee();
```

Comment:
I don't think we need to care how much money an account has left. If the account has a lot of money but is willing to pay up to the offered fee, then that is its own decision.

Reply:
These methods allow the `FeeChargingPolicy` to be unaware of _how_ to get the payer balance; but it needs these three pieces of information to decide whether to return `INSUFFICIENT_TX_FEE` or `INSUFFICIENT_PAYER_BALANCE`.